pallets/verifiers/src/lib.rs

```diff
@@ -293,6 +348,35 @@ pub mod pallet {
             Disabled::<T, I>::put(disabled);
             Ok(())
         }
+
+        /// Unregister a previously registered verification key.
+        /// Should be called by the same account used for registering the verification key.
+        /// Unlock the funds which were locked when registering the verification key.
+        #[pallet::call_index(3)]
+        #[pallet::weight(T::WeightInfo::unregister_vk())]
+        pub fn unregister_vk(origin: OriginFor<T>, vk_hash: H256) -> DispatchResult {
+            log::trace!("Unregister vk");
+            let account_id = ensure_signed(origin)?;
+            if let Some(ticket) = Tickets::<T, I>::take((&account_id, vk_hash)) {
+                ticket.drop(&account_id)?;
+                Vks::<T, I>::mutate_exists(vk_hash, |vk_entry| match vk_entry {
+                    Some(v) => {
+                        if v.ref_count > 1 {
+                            v.ref_count -= 1;
+                        } else {
+                            *vk_entry = None;
+                        }
```

Comment:
consider to use always `saturete_sub(1)` instead and the check if `v.ref_count == 0`